blockchain/validate.go

```diff
@@ -277,172 +321,119 @@ func checkTransactionSanityContextual(tx *wire.MsgTx, params *chaincfg.Params, i
 		// Check script length of stake base signature.
 		slen := len(tx.TxIn[0].SignatureScript)
 		if slen < MinCoinbaseScriptLen || slen > MaxCoinbaseScriptLen {
-			str := fmt.Sprintf("stakebase transaction script "+
-				"length of %d is out of range (min: %d, max: "+
-				"%d)", slen, MinCoinbaseScriptLen,
+			str := fmt.Sprintf("stakebase transaction script length of %d is "+
+				"out of range (min: %d, max: %d)", slen, MinCoinbaseScriptLen,
 				MaxCoinbaseScriptLen)
 			return ruleError(ErrBadStakebaseScriptLen, str)
 		}
 
 		// The script must be set to the one specified by the network.
-		// Check script length of stake base signature.
-		if !bytes.Equal(tx.TxIn[0].SignatureScript,
-			params.StakeBaseSigScript) {
-			str := fmt.Sprintf("stakebase transaction signature "+
-				"script was set to disallowed value (got %x, "+
-				"want %x)", tx.TxIn[0].SignatureScript,
-				params.StakeBaseSigScript)
+		if !bytes.Equal(tx.TxIn[0].SignatureScript, params.StakeBaseSigScript) {
+			str := fmt.Sprintf("stakebase transaction signature script was "+
+				"set to disallowed value (got %x, want %x)",
+				tx.TxIn[0].SignatureScript, params.StakeBaseSigScript)
 			return ruleError(ErrBadStakebaseScrVal, str)
 		}
 
 		// The ticket reference hash in an SSGen tx must not be null.
 		ticketHash := &tx.TxIn[1].PreviousOutPoint
 		if isNullOutpoint(ticketHash) {
-			return ruleError(ErrBadTxInput, "ssgen tx ticket input"+
-				" refers to previous output that is null")
+			str := "vote ticket input refers to previous output that is null"
+			return ruleError(ErrBadTxInput, str)
```

Comment:
I believe this is now impossible to hit (and is the reason that one test needed its error value changed) since the `checkTransactionSanityContextFree()` now already ensures votes can't have two inputs with an empty outpoint but I'm guessing you're keeping it as future proofing in case the previous check ever changes.

Reply:
Also, this change doesn't cause a potential hard fork since it's just changing one error for another and nothing is relying on a _specific_ error value for this situation.

Reply:
Correct on both accounts.  I contemplated removing it, but I figured it would be nice to keep for future proofing in case things move around again.